src/node_file.cc

```diff
@@ -599,8 +590,17 @@ static void FTruncate(const FunctionCallbackInfo<Value>& args) {
 
   int fd = args[0]->Int32Value();
 
-  ASSERT_TRUNCATE_LENGTH(args[1]);
-  int64_t len = GET_TRUNCATE_LENGTH(args[1]);
+  // FIXME(bnoordhuis) It's questionable to reject non-ints here but still
+  // allow implicit coercion from null or undefined to zero.  Probably best
+  // handled in lib/fs.js.
```

Comment:
agreed. it's much easier to handle value coercion from JS. even if that means all there is is a minor wrapper that does a couple coercions and checks.